File: 2022/aoc_python/day_1.py
from lib import io

print("Advent of Code 2022 Day 1")
print("-------------------------")
# Load calorie counts.
calorie_log_file = "input/day-1-input.txt"
calorie_counts = io.file_to_list(calorie_log_file)

# Get total calories carried by each elf.
total_calories = []
calorie_sum = 0
for calorie_count in calorie_counts:
    if calorie_count:
        calorie_sum += int(calorie_count)
    else:
        total_calories.append(calorie_sum)
        calorie_sum = 0

# Get highest calorie count.
total_calories.sort()
print("PART 1")
print("======")
print(f"Highest calorie count: {total_calories[0]}")
print("======")

# Get sum of top three calorie counts.
print("PART 2")
print("======")
top_three_calorie_sum = total_calories.pop() + total_calories.pop() + total_calories.pop()
print(f"Top three calorie sum: {top_three_calorie_sum}")

File: 2022/aoc_perl/day_1.pl
#!/usr/bin/perl
use strict;
use warnings;

use FindBin;                                # Useful for finding directory where script is being invoked.
use lib "$FindBin::Bin/../aoc_perl/lib";    # Identifies library folder.

use util qw(file_to_array println);

print("Advent of Code 2022 Day 1\n");
print("-------------------------\n");

# Load calorie counts.
my @calorie_counts = file_to_array(
    PATH_TO_FILE => "$FindBin::Bin/../input",
    FILE_NAME => "day-1-input.txt"
);
# print("calorie_counts: @calorie_counts\n");

# Get total calories carried by each elf.
my @total_calories = ();
my $calorie_sum = 0;
for my $calorie_count (@calorie_counts) {
    # print("calorie_count: $calorie_count\n");
    if ($calorie_count) {
        # print("Truthy.\n");
        $calorie_sum += int($calorie_count);
        # print("calorie_sum: $calorie_sum\n");
    } else {
        # print("Falsey.\n");
        unshift(@total_calories, $calorie_sum);
        # print("total_calories: @total_calories\n");
        $calorie_sum = 0;
    }
}

# Get highest calorie count.
@total_calories = sort { $b <=> $a } @total_calories;   # Sort numerically in descending order.
print("PART 1\n");
print("======\n");
print("Highest calorie count: $total_calories[0]\n");
print("======\n");

# Get sum of top three calorie counts.
print("PART 2\n");
print("======\n");
my $top_three_calorie_sum = $total_calories[0] + $total_calories[1] + $total_calories[2];
print("Top three calorie sum: $top_three_calorie_sum\n");
